generators/app/gulpfile.js

```diff
@@ -10,56 +10,54 @@
 
 // Here we add tools that will be used to process our source files.
+const babel = require('gulp-babel');
+const babelPresetES2015 = require('babel-preset-es2015');
+const babiliPreset = require('babel-preset-babili');
+const cssSlam = require('css-slam').gulp;
+const externalHelpersPlugin = require('babel-plugin-external-helpers');
+const htmlMinifier = require('gulp-html-minifier');
 const imagemin = require('gulp-imagemin');
 
-// Additional plugins can be used to optimize your source files after splitting.
-// Before using each plugin, install with `npm i --save-dev <package-name>`
-// const uglify = require('gulp-uglify');
-// const cssSlam = require('css-slam').gulp;
-// const htmlMinifier = require('gulp-html-minifier');
-
-const swPrecacheConfig = require('./sw-precache-config.js');
-const polymerJson = require('./polymer.json');
-const polymerProject = new polymerBuild.PolymerProject(polymerJson);
-const buildDirectory = 'build';
-
-/**
- * Waits for the given ReadableStream
- */
-function waitFor(stream) {
-  return new Promise((resolve, reject) => {
-    stream.on('end', resolve);
-    stream.on('error', reject);
-  });
-}
+// Build out the Polymer Project Config instance
+const polymerProject = getProjectConfig(require('./polymer.json'), getArgs());
 
-function build() {
-  return new Promise((resolve, reject) => { // eslint-disable-line no-unused-vars
+function runBuilds() {
+  const mainBuildDirectoryName = 'build';
+  const builds = polymerProject.config.builds || [];
 
-    // Lets create some inline code splitters in case you need them later in your build.
-    let sourcesStreamSplitter = new polymerBuild.HtmlSplitter();
-    let dependenciesStreamSplitter = new polymerBuild.HtmlSplitter();
+  // Okay, so first thing we do is clear the build directory
+  console.log(`Clearing ${mainBuildDirectoryName}${path.sep} directory...`);
+  return del([mainBuildDirectoryName])
+    .then(() => {
+      return Promise.all(builds.map((options) => {
+        // If no name is provided, write directly to the build/ directory.
+        // If a build name is provided, write to that subdirectory.
+        const buildName = options.name || 'default';
+        const buildDirectory = path.join(mainBuildDirectoryName, buildName);
 
-    // Okay, so first thing we do is clear the build directory
-    console.log(`Deleting ${buildDirectory} directory...`);
-    del([buildDirectory])
-      .then(() => {
+        return new Promise((resolve) => {
+          console.log(`(${buildName}) Building...`);
 
-        // Let's start by getting your source files. These are all the files
-        // in your `src/` directory, or those that match your polymer.json
-        // "sources"  property if you provided one.
-        let sourcesStream = polymerProject.sources()
+          // Lets create some inline code splitters in case you need them later in your build.
+          const sourcesStreamSplitter = new polymerBuild.HtmlSplitter();
```

Comment:
I think I would just use define splitter and add a commented out section of code showing how to wire a splitter into the sources stream.